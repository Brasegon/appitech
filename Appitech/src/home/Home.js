import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  StatusBar,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

  
export default class Home extends React.Component {     
  render() {
    return (
      <View style={styles.container}>

              <View style={{ backgroundColor: '#bbe1fa' }}>
        <View style={styles.container2}>
        <Image source={require('../../Asset/Appitech.png')} style={{ width: 230, height: 60, top:20, left: 80 }} />

            </View>
            </View>



        <View style={{ backgroundColor: '#0f4c75' }}>
                  <LinearGradient colors={['#bbe1fa', '#3282b8']} style={styles.container1}>
                <View style={{top:-25}}>
                    <Text style={styles.title}>Prochaine activité</Text>
                    <View style={{backgroundColor:"#F2F1F7",
                                  width:350, left:30, top:60, borderBottomLeftRadius: 30,
                                  borderBottomRightRadius: 30,
                                  borderTopRightRadius: 30,
            borderTopLeftRadius: 30, height: 100}}>
      <MaterialCommunityIcons name="calendar-blank-multiple" color={"#0f4c75"} size={50} style={{position:"absolute", top:20, left:20}} /> 
                    <Text style={{fontSize:17, left:90, top:15, width:250, paddingBottom:5}}>Review projet de fin d'année</Text>
                    <Text style={{fontSize:13, left:90, top:15, width:250, color:'grey'}}>Salle Scifi <MaterialCommunityIcons name="sign-direction" color={"grey"} size={15} /></Text>
                    <Text style={{fontSize:13, left:90, top:15, width:250 , color:'grey'}}>11h30 <MaterialCommunityIcons name="clock-time-eight-outline" color={"grey"} size={15} /></Text>

                    </View>
                    </View>
                  </LinearGradient>
        </View>

    <View style={{ backgroundColor: '#231332' }}>
        <View style={styles.container3}>
            <View style={{top:-30}}>
        <Text style={styles.title}>Projets en cours</Text>
        <View style={{backgroundColor:"#F2F1F7",
                                  width:350, left:30, top:60, borderBottomLeftRadius: 30,
                                  borderBottomRightRadius: 30,
                                  borderTopRightRadius: 30,
            borderTopLeftRadius: 30, height: 140}}>
      <MaterialCommunityIcons name="folder-open-outline" color={"#0f4c75"} size={50} style={{position:"absolute", top:20, left:20}} /> 
                    <Text style={{fontSize:17, left:90, top:15, width:250, paddingBottom:5}}>Epicture</Text>
                    <Text style={{fontSize:17, left:90, top:15, width:250, paddingBottom:5}}>E-Commerce</Text>
                    <Text style={{fontSize:17, left:90, top:15, width:250, paddingBottom:5}}>My_FTP</Text>
                    <Text style={{fontSize:17, left:90, top:15, width:250, paddingBottom:5}}>Minishell2</Text>

                    </View>
                    </View>
                    <View style={{marginTop:20}}>
                    <Text style={styles.title}>Dernières notes</Text>
        <View style={{backgroundColor:"#F2F1F7",
                                  width:350, left:30, top:60, borderBottomLeftRadius: 30,
                                  borderBottomRightRadius: 30,
                                  borderTopRightRadius: 30,
            borderTopLeftRadius: 30, height: 90}}>
      <MaterialCommunityIcons name="checkbox-marked-outline" color={"#0f4c75"} size={50} style={{position:"absolute", top:20, left:20}} /> 
                    <Text style={{fontSize:17, left:90, top:15, width:250, paddingBottom:5}}>Devops : <Text style={{color : "#0f4c75"}}>A</Text></Text>
                    <Text style={{fontSize:17, left:90, top:15, width:250, paddingBottom:5}}>Java : <Text style={{color : "#0f4c75"}}>B</Text></Text>
                    </View>
                    </View>

        </View>
    </View>


      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  container1: {
    height: 230,
    padding: 10,
    borderBottomLeftRadius:100,
  },
  container2: {
    backgroundColor: 'white',
    height: 130,
    padding: 10,
    borderBottomLeftRadius:100,
  },
  container3: {
      backgroundColor : '#0f4c75',
    height: 600,
    padding: 10,
    borderBottomLeftRadius:100,
  },
  title: {
      fontSize: 20,
      color : 'white',
      left : 50,
      top : 45

  }
});