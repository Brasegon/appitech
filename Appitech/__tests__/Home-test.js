/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import Home from '../src/home/Home';
 import { SearchBar, Overlay } from 'react-native-elements';
 import {shallow} from 'enzyme';
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
import SearchBarHome from '../src/home/searchBar/searchBar';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
configure({ adapter: new Adapter() });

 it('Render Home', () => {
   const tree = renderer.create(<Home/>).toJSON();
   expect(tree).toMatchSnapshot();
 })

 describe("Search Bar", () => {
  it('Search is functionnal', () => {
    const wrapper = shallow(<SearchBarHome search="Epicture Test"/>);
    const instance = wrapper.instance();
    var test = wrapper.find(SearchBar);
    test.simulate('keypress', {key: 'Enter'});
    wrapper.update();
    expect(test.props().value).toEqual("Epicture Test");
  });
});