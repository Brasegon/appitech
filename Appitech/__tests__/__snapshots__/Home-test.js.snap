// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home 1`] = `
<View>
  <View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, .4)",
            "height": "100%",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
        testID="RNE__Overlay__backdrop"
      />
      <View
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 3,
              "elevation": 2,
              "padding": 10,
            }
          }
        >
          <View>
            <ActivityIndicator
              animating={true}
              color="#0000ff"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomColor": "#e1e1e1",
          "borderBottomWidth": 1,
          "borderRadius": 0,
          "borderTopColor": "#e1e1e1",
          "borderTopWidth": 1,
          "height": 60,
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 15,
              "flexDirection": "row",
              "height": 40,
              "minHeight": 30,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#86939e",
                          "fontSize": 18,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    ï€‚
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Rechercher une image"
            placeholderTextColor="#86939e"
            platform="default"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#86939e",
                "flex": 1,
                "fontSize": 18,
                "marginLeft": 10,
                "minHeight": 40,
              }
            }
            testID="searchInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginRight": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "height": 0,
              "margin": 0,
              "padding": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View>
    <RCTScrollView
      disableVirtualization={false}
      extraData={null}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "left": 130,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../Asset/Imgur.png",
        }
      }
      style={
        Object {
          "height": 200,
          "left": -50,
          "top": 100,
          "width": 200,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#cdcdcd",
          "fontSize": 20,
          "left": -70,
          "top": 120,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#D20073",
          }
        }
      >
        Launch
      </Text>
       a search to access
      

         the entire Imgur library !
    </Text>
  </View>
</View>
`;
